### PR TITLE
Build/css chunking

### PR DESCRIPTION
**Pull request by benjag**
Credit to Kurt for coming up with plugin technique, the branch was pretty stale (and pre TS) so I decided to manually bring his work in. See https://github.com/rei/rei-cedar/pull/141

### CODE REVIEW

File: docgen.mjs [hunk 1]
```diff
@@ -21,7 +21,7 @@ await Promise.all(iconFiles.map(async (filePath) => {
 }))
 
 async function createDocgenObj(filePath, docgenObj) {
-    let parsedComponentFile = await parse(filePath);
+    const parsedComponentFile = await parse(filePath);
```
Comment (benjag) [inline docgen.mjs:24]: Never reassigned

File: docgen.mjs [hunk 2]
```diff
@@ -49,8 +49,6 @@ function trimApostrophes(str) {
     return str;
 }
 
-const components = Object.keys(componentObj); // Replace with your actual component names
```
Comment (benjag) [inline docgen.mjs:52]: Don't believe this was actually doing anything. Lint was flagging as unused declaration

File: package.json [hunk 3]
```diff
@@ -88,18 +88,12 @@
...
-    "node-sass-package-importer": "^5.3.2",
     "npm-publish-prerelease": "^1.0.2",
     "npm-run-all": "^4.1.5",
-    "postcss": "^8.3.0",
+    "postcss": "^8.4.31",
     "postcss-calc": "^8.0.0",
-    "postcss-import": "^14.0.2",
-    "postcss-inline-svg": "^5.0.0",
-    "postcss-load-config": "^3.0.1",
-    "postcss-modules": "^4.0.0",
```
Comment (benjag) [inline package.json:100]: 🤤

File: src/components/card/styles/vars/CdrCard.vars.scss [hunk 4]
```diff
@@ -6,6 +6,7 @@
   box-shadow: $cdr-prominence-raised;
   //ITEM_DOC: Text color of a cdr-card
   color: var(--cdr-card-base-color, var(--cdr-color-text-primary, #{$cdr-color-text-primary}));
+  container: card / inline-size;
```
Comment (benjag) [inline src/components/card/styles/vars/CdrCard.vars.scss:9]: Just declaring that card is a container for future use

File: src/components/text/presets/textPresets.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
-export { default as CdrEyebrow } from './PresetEyebrow.vue';
\ No newline at end of file
+export { default as CdrHeadingDisplay } from './CdrHeadingDisplay.vue';
+export { default as CdrHeadingSerif } from './CdrHeadingSerif.vue';
+export { default as CdrHeadingSans } from './CdrHeadingSans.vue';
+export { default as CdrSubheadingSans } from './CdrSubheadingSans.vue';
+export { default as CdrUtilitySans } from './CdrUtilitySans.vue';
+export { default as CdrUtilitySerif } from './CdrUtilitySerif.vue';
+export { default as CdrBody } from './CdrBody.vue';
+export { default as CdrEyebrow } from './CdrEyebrow.vue';
```
Comment (benjag) [inline src/components/text/presets/textPresets.ts:8]: Vite was naming the CSS files according to the native component name so dropping this naming convention

File: src/components/button/styles/vars/CdrButton.vars.scss [hunk 6]
```diff
@@ -2,18 +2,18 @@
   font-family: $cdr-font-family-sans;
   font-style: normal;
   font-weight: 500;
-  letter-spacing: -.08px;
-  font-size: 14px;
-  line-height: 18px;
+  letter-spacing: -.008rem;
+  font-size: 1.4rem;
```
Comment (mhewson) [inline src/components/button/styles/vars/CdrButton.vars.scss:6]: do you know why there were px in the first place?

Comment (mhewson) [reply src/components/button/styles/vars/CdrButton.vars.scss:6]: I'm actively refactoring this file to remove as many of the mixins as we can

**Review by mhewson — APPROVED** [review]

File: src/components/button/styles/vars/CdrButton.vars.scss (hunk 6 above)
Comment (benjag) [reply src/components/button/styles/vars/CdrButton.vars.scss:6]: I don't, it's been that way for 6 years though so maybe just old

**Pull request merged**